scripts/o3de/o3de/export_project.py

```diff
@@ -7,28 +7,94 @@
+    O3DE_SCRIPT_NAME = 'o3de.bat'
+    GENERATOR = None
+    CMAKE_GENERATOR_OPTIONS = ['-DLY_DISABLE_TEST_MODULES=ON']
+    CMAKE_MULTI_CONFIGURATION_GENERATOR = True
+    ADDITIONAL_PLATFORM_IGNORE_FILES = ['*.pdb', '*.lock']
+
+else:
+    # Test if Ninja is available from the command line to determine the generator and multi-config capability
+    test_ninja_result = None
+    try:
+        test_ninja_result = subprocess.run(['ninja', '--version'])
+    except FileNotFoundError:
+        pass
+
+    if test_ninja_result and test_ninja_result.returncode == 0:
+        GENERATOR = "Ninja Multi-Config"
+        CMAKE_MULTI_CONFIGURATION_GENERATOR = True
+    else:
+        GENERATOR = "Unix Makefiles"
+        CMAKE_MULTI_CONFIGURATION_GENERATOR = False
+
+    EXECUTABLE_EXTENSION = ""
+    O3DE_SCRIPT_NAME = 'o3de.sh'
+    CMAKE_GENERATOR_OPTIONS = ['-DLY_DISABLE_TEST_MODULES=ON', '-DLY_STRIP_DEBUG_SYMBOLS=ON']
+    ADDITIONAL_PLATFORM_IGNORE_FILES = ['*.dbg', '*.lock']
+
+CUSTOM_SCRIPT_HELP_ARGUMENT = '--script-help'
+
+
+# Regardless of the output package configuration, the tools used for the export process must be built with
+# the profile configuration
+PREREQUISITE_TOOL_BUILD_CONFIG = "profile"
+
+
+class ExportProjectError(RuntimeError):
+    """
+    Define an error related to the runtime execution of export project so to handle messaging properly
+    """
+    pass
+
 
 class O3DEScriptExportContext(object):
     """
     The context object is used to store parameter values and variables throughout the lifetime of an export script's execution.
     It can also be passed onto nested scripts the export script may execute, which can in turn update the context as necessary.
     """
     
-    def __init__(self, export_script_path: pathlib.Path,
-                       project_path: pathlib.Path,
-                       engine_path: pathlib.Path,
-                       args: list = []) -> None:
+    def __init__(self,
+                 export_script_path: pathlib.Path,
+                 project_path: pathlib.Path,
+                 engine_path: pathlib.Path,
+                 args: list = [],
+                 cmake_additional_build_args: list = []) -> None:
+
         self._export_script_path = export_script_path
         self._project_path = project_path
         self._engine_path = engine_path
         self._args = args
+        self._cmake_additional_build_args = cmake_additional_build_args
+
+        project_json_data = manifest.get_project_json_data(project_path=project_path)
+        assert project_json_data, f"Invalid project configuration file '{project_path}/project.json'. Invalid settings."
+
+        project_name = project_json_data.get('project_name')
+        assert project_name, f"Invalid project configuration file '{project_path}/project.json'. 'project_name' not found in the settings"
+
+        self._project_name = project_name
+        self._is_engine_centric = project_path.is_relative_to(engine_path)
```

Comment:
What it means to be engine centric vs project-centric is what is used as the root CMakeLists.txt for cmake configuration?

For example if you configured with the AutomatedTesting directory as the source directory for cmake, then that is project-centric, since it uses the project CMakeLists.txt for as the cmake entry point.

```bash
[spham-amzn@ubuntu ~/o3de]$ cmake -B AutomatedTesting/build/linux -S AutomatedTesting
```

If you use the engine directory as the cmake source directory, then that is engine-centric.
The initial cmake entry in that case is the CMakeLists.txt in the root of the engine
```bash
[spham-amzn@ubuntu ~/o3de]$ cmake -B AutomatedTesting/build/linux -S . -DLY_PROJECT=./AutomatedTesting
```

What this means is that unless you ask the user for whether they want to use the project root or engine root as the Top level cmake source directory, then the code is just guessing at which one to use.

Reply:
If that is the case, and its possible to set `-DLY_PROJECT` to an absolute path, then its probably better to make an engine build a switch (argument) that is passed in instead of assuming a project folder outside of the engine root is project centric, etc.

Reply:
Yes, `LY_PROJECT` accepts a path that can be absolute or relative to the [cmake source directory](https://github.com/o3de/o3de/blob/development/cmake/Projects.cmake#L263).
With the information that the script has, it can always choose either project-centric or engine-centric in every situation.